sshd-core/src/main/java/org/apache/sshd/common/config/keys/OpenSSHKeyUtils.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sshd.common.config.keys;
+
+import java.security.PublicKey;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.sshd.common.Factory;
+import org.apache.sshd.common.digest.BuiltinDigests;
+import org.apache.sshd.common.digest.Digest;
+import org.apache.sshd.common.digest.DigestFactory;
+import org.apache.sshd.common.util.Base64;
+import org.apache.sshd.common.util.Pair;
+import org.apache.sshd.common.util.ValidateUtils;
+import org.apache.sshd.common.util.buffer.Buffer;
+import org.apache.sshd.common.util.buffer.BufferUtils;
+import org.apache.sshd.common.util.buffer.ByteArrayBuffer;
+
+/**
+ * Utility class for keys
+ *
+ * @author <a href="mailto:dev@mina.apache.org">Apache MINA SSHD Project</a>
+ */
+public final class OpenSSHKeyUtils {
+
+    /**
+     * The default {@link Factory} of {@link Digest}s initialized
+     * as the value of {@link #getDefaultFingerPrintFactory()}
+     */
+    public static final Factory<Digest> DEFAULT_FINGERPRINT_DIGEST_FACTORY = BuiltinDigests.sha256;
+
+    private static final AtomicReference<Factory<? extends Digest>> DEFAULT_DIGEST_HOLDER =
+            new AtomicReference<Factory<? extends Digest>>(DEFAULT_FINGERPRINT_DIGEST_FACTORY);
+
+    private OpenSSHKeyUtils() {
+        throw new UnsupportedOperationException("No instance");
+    }
+
+    /**
+     * @return The default {@link Factory} of {@link Digest}s used
+     * by the {@link #getFingerPrint(PublicKey)} and {@link #getFingerPrint(String)}
+     * methods
+     * @see #setDefaultFingerPrintFactory(Factory)
+     */
+    public static Factory<? extends Digest> getDefaultFingerPrintFactory() {
+        return DEFAULT_DIGEST_HOLDER.get();
+    }
```

Comment:
If you really want to be flexible you can make this configurable (I am willing to accept the code even without this)
```java
/**
  * System property that can be used to control the default fingerprint factory used for keys.
  * If not set the {@link #DEFAULT_FINGERPRINT_DIGEST_FACTORY} is used
  */
public static final String OPENSSH_KEY_FINGERPRINT_FACTORY_PROP = "org.apache.sshd.opensshFingerprintFactory";

public static Factory<? extends Digest> getDefaultFingerPrintFactory() {
    Factory<? extends Digest> factory = null;
    synchronized(DEFAULT_DIGEST_HOLDER) {
        factory = DEFAULT_DIGEST_HOLDER.get();
        if (factory != null) {
            return factory;
        }

        String propVal = System.getProperty(OPENSSH_KEY_FINGERPRINT_FACTORY_PROP);
        if (GenericUtils.isEmpty(propVal)) {
            factory = DEFAULT_FINGERPRINT_DIGEST_FACTORY;
        } else {
            factory = ValidateUtils.checkNotNull(BuiltinDigests.fromFactoryName(propVal), "Unknown digest factory: %s", propVal);
        }

        DEFAULT_DIGEST_HOLDER.set(factory);
    }

    return factory
}

```